### PR TITLE
fix: Add explicit labels (using: for & id) to input fields 

### LINKED ISSUE #6576: Accessibility [1.3.1]: Labels for login/register and reset password must be explicitly associated (with "for" and "id")
### Is your feature request related to a problem? Please describe.

This issue was reported by the Accessibility Foundation (https://github.com/openfoodfacts/openfoodfacts-server/issues/4640) in the [Accessibility_report.pdf](https://github.com/openfoodfacts/openfoodfacts-server/files/5702582/Report.OpenFoodFact.15-12-2020.pdf) [read: 1.3.1]

Implicit (`<label>` Username or password `<input>` text `</input>` `</label>`) is insufficient for the screen readers to process.

### Describe the solution you'd like

Explicit labels should be implemented for all input fields

### PR DESCRIPTION
**Pull request by yuktea**
### Description
This change makes the experience of using a screen reader better for visually impaired people. (prescribed in the [Accessibility Report](https://github.com/openfoodfacts/openfoodfacts-server/files/5702582/Report.OpenFoodFact.15-12-2020.pdf))
This ensures that assistive technology can refer to the correct label when presenting a form control.

### Related issue(s) and discussion
- Fixes #6576
- A part of this issue thread https://github.com/openfoodfacts/openfoodfacts-server/issues/4640

### CODE REVIEW

File: templates/web/pages/reset_password/reset_password.tt.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
     [% INCLUDE 'web/common/includes/error_list.tt.html' %]
     <form method="post" action="/cgi/reset_password.pl" enctype="multipart/form-data">
         [% IF type == 'send_email' %]
-            <label>
+            <label for="username or email">
```
Comment (VaiTon) [inline templates/web/pages/reset_password/reset_password.tt.html:8]: I don't think this is a correct id

**Review by VaiTon — CHANGES_REQUESTED** [review]
Thank you for your work! Just some changes here and there!

File: templates/web/pages/reset_password/reset_password.tt.html (hunk 1 above)
Comment (yuktea) [reply templates/web/pages/reset_password/reset_password.tt.html:8]: > 

Yes @VaiTon I was unsure of this. I think it should be `userid_or_email`

Comment (VaiTon) [reply templates/web/pages/reset_password/reset_password.tt.html:8]: I agree. You should add it also to the input under this line!

File: templates/web/common/includes/display_login_register.tt.html [hunk 2]
```diff
@@ -5,17 +5,17 @@
...
-			<label>[% lang('password') %]
-				<input type="password" name="password" autocomplete="current-password" required>
+			<label for="password" >[% lang('password') %]
+				<input type="password" name="password" id="password" autocomplete="current-password" required>
 			</label>
 		</div>
 		<div class="small-12 columns">
-			<label>
+			<label for="remember_me_checkbox">
 				<input type="checkbox" name="remember_me" checked>
```
Comment (VaiTon) [inline templates/web/common/includes/display_login_register.tt.html:19]: Here there is no id

File: templates/web/pages/change_password/change_password.tt.html [hunk 3]
```diff
@@ -5,34 +5,34 @@
...
             </div>
         </div>
 
         <div class="row">
             <div class="large-12 columns">
-                <label>
+                <label for="new_password">
                     [% lang('password_confirm') %]
-                    <input type="password" name="confirm_password" autocomplete="new-password" required="required" />
+                    <input type="password" name="confirm_password" id="new_password" autocomplete="new-password" required="required" />
```
Comment (VaiTon) [inline templates/web/pages/change_password/change_password.tt.html:28]: No two fields with the same id. Please use `new_password_confirm`

File: templates/web/pages/change_password/change_password.tt.html [hunk 4]
```diff
@@ -5,34 +5,34 @@
...
                 </label>
             </div>
         </div>
 
         <div class="row">
             <div class="large-12 columns">
-                <label>
+                <label for="submit_form">
                     <input type="submit" name="submit" class="button" />
```
Comment (VaiTon) [inline templates/web/pages/change_password/change_password.tt.html:36]: No id here

File: templates/web/pages/login_form/login.tt.html [hunk 5]
```diff
@@ -23,7 +23,7 @@
 
     <div class="row">
         <div class="large-12 columns">
-            <label>
+            <label for="submit">
                 <input type="submit" name="submit" class="button" />
```
Comment (VaiTon) [inline templates/web/pages/login_form/login.tt.html:27]: No id here

**Review by VaiTon — CHANGES_REQUESTED** [review]

**Review by VaiTon — APPROVED** [review]
Thank you! 🎉

**Pull request merged**